envoy/router/path_match.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Decides if the target route path matches the provided pattern.
+ * Subclassing Logger::Loggable so that implementations can log details.
+ */
+class PathMatcher : Logger::Loggable<Logger::Id::router> {
+public:
+  PathMatcher() = default;
+  virtual ~PathMatcher() = default;
+
+  /**
+   * Determines if the current path matches the pattern.
+   *
+   * @param path current path of the matchable route
+   * @return true if path matches the pattern.
+   */
+  virtual bool match(absl::string_view path) const PURE;
+
+  /**
+   * @return the match pattern.
```

Comment:
Here and in the Rewriter, what format is this string in? Is it an re2 regex or something else?

Reply:
It would depend on the implementation

Reply:
Hm. Is this method simply there for debugging output, or is its contents consumed programmatically somewhere?

Reply:
Not sure what the needed fix is here. Is there a change needed?

Reply:
From your reply that the format of this return value is implementation dependent, I would expect that this value is only used in an implementation-specific fashion (or just logged). So I traced the code to see where this value is used, and it shows up in the code I mentioned. I'm trying to understand if that code is correct or not?